src/libraries/Microsoft.PowerFx.Core/Public/Types/Serialization/FormulaTypeSchema.cs

```diff
@@ -40,6 +39,7 @@ public enum ParamType
         /// <summary>
         /// Optional. For Records and Tables, contains the list of fields.
         /// </summary>
+        [SuppressMessage("Usage", "CA2227: Collection properties should be read only", Justification = "n/a")]
```

Comment:
>[SuppressMessage("Usage", "CA2227: Collection properties should be read only", Justification = "n/a")] [](http://example.com/codeflow?start=8&length=102)

Can we make it a readonlydictionary instead? It's set on FormulaTypeToSchemaConverter, but only when creating a new FormulaTypeSchema, so we can make it part of hte .ctor

Reply:
This causes some trouble to Json desialization, so I'd avoid this change for now